testsuite/model/src/test/java/org/keycloak/testsuite/model/infinispan/PartitionManager.java

```diff
@@ -0,0 +1,261 @@
+            )
+            .collect(Collectors.toList());
+   }
+
+   private JChannel channel(int index) {
+      return TestingUtil.extractJChannel(cacheManagers[index]);
+   }
+
+   private static class Partition {
+      final AtomicInteger viewId;
+      final List<Address> allMembers;
+      final List<Cache<?, ?>> caches;
+      final List<JChannel> channels = new ArrayList<>();
+
+      public Partition(AtomicInteger viewId, List<Address> allMembers, List<Cache<?, ?>> caches) {
+         this.viewId = viewId;
+         this.allMembers = allMembers;
+         this.caches = caches;
+      }
+
+      public void addNode(JChannel c) {
+         channels.add(c);
+      }
+
+      public void partition() {
+         log.trace("Partition forming");
+         disableDiscovery();
+         installNewView();
+         assertPartitionFormed();
+         log.trace("New views installed");
+      }
+
+      private void disableDiscovery() {
+         channels.forEach(c ->
+               ((Discovery) c.getProtocolStack().findProtocol(Discovery.class)).setClusterName(c.getAddressAsString())
+         );
+      }
+
+      private void assertPartitionFormed() {
+         final List<Address> viewMembers = new ArrayList<>();
+         for (JChannel ac : channels) viewMembers.add(ac.getAddress());
+         for (JChannel c : channels) {
+            List<Address> members = c.getView().getMembers();
+            if (!members.equals(viewMembers)) throw new AssertionError();
+         }
+      }
+
+      private List<Address> installNewView() {
+         final List<Address> viewMembers = new ArrayList<>();
+         for (JChannel c : channels) viewMembers.add(c.getAddress());
+         View view = View.create(channels.get(0).getAddress(), viewId.incrementAndGet(),
+               viewMembers.toArray(new Address[0]));
+
+         log.trace("Before installing new view...");
+         for (JChannel c : channels) {
+            getGms(c).installView(view);
+         }
+         return viewMembers;
+      }
+
+      private List<Address> installMergeView(ArrayList<JChannel> view1, ArrayList<JChannel> view2) {
+         List<Address> allAddresses =
+               Stream.concat(view1.stream(), view2.stream()).map(JChannel::getAddress).distinct()
+                     .collect(Collectors.toList());
+
+         View v1 = toView(view1);
+         View v2 = toView(view2);
+         List<View> allViews = new ArrayList<>();
+         allViews.add(v1);
+         allViews.add(v2);
+
+         // Remove all sent NAKACK2 messages to reproduce ISPN-9291
+         for (JChannel c : channels) {
+            STABLE stable = c.getProtocolStack().findProtocol(STABLE.class);
+            stable.gc();
+         }
+         try {
+            Thread.sleep(10);
+         } catch (InterruptedException e) {
+            e.printStackTrace();
+         }
+
+         MergeView mv = new MergeView(view1.get(0).getAddress(), viewId.incrementAndGet(), allAddresses, allViews);
```

Comment:
The view id should be the highest view id +1

https://github.com/belaban/JGroups/blob/1f6da1af5fffda1f5b7e52a9f96575e08f5dd693/src/org/jgroups/protocols/pbcast/Merger.java#L716-L717

Reply:
I think that will be the case as we're calling `viewId.incrementAndGet` each time when we do:

```java
         View v1 = toView(view1);
         View v2 = toView(view2); 
```

This is the same code that's in BasePartitionHandlingTest, so if there's an issue we need to fix there as well.

Reply:
ah, that sounds good to me then.